### ISSUE TITLE
Error with build C

### ISSUE THREAD

**Issue by abersheeran** [pdm-project/pdm-pep517#102]
```
pdm.termui: Traceback (most recent call last):
pdm.termui:   File "/opt/hostedtoolcache/Python/3.8.13/x64/lib/python3.8/site-packages/pep517/in_process/_in_process.py", line 363, in <module>
pdm.termui:     main()
pdm.termui:   File "/opt/hostedtoolcache/Python/3.8.13/x64/lib/python3.8/site-packages/pep517/in_process/_in_process.py", line 345, in main
pdm.termui:     json_out['return_val'] = hook(**hook_input['kwargs'])
pdm.termui:   File "/opt/hostedtoolcache/Python/3.8.13/x64/lib/python3.8/site-packages/pep517/in_process/_in_process.py", line 283, in build_editable
pdm.termui:     return hook(wheel_directory, config_settings, metadata_directory)
pdm.termui:   File "/tmp/pdm-build-env-85zkiyg_-shared/lib/python3.6/site-packages/pdm/pep517/api.py", line 98, in build_editable
pdm.termui:     return Path(builder.build(wheel_directory)).name
pdm.termui:   File "/tmp/pdm-build-env-85zkiyg_-shared/lib/python3.6/site-packages/pdm/pep517/wheel.py", line 75, in build
pdm.termui:     self._build(zip_file)
pdm.termui:   File "/tmp/pdm-build-env-85zkiyg_-shared/lib/python3.6/site-packages/pdm/pep517/editable.py", line 79, in _build
pdm.termui:     setup_py = self.ensure_setup_py()
pdm.termui:   File "/tmp/pdm-build-env-85zkiyg_-shared/lib/python3.6/site-packages/pdm/pep517/base.py", line 410, in ensure_setup_py
pdm.termui:     setup_py_path.write_text(self.format_setup_py(), encoding="utf-8")
pdm.termui:   File "/tmp/pdm-build-env-85zkiyg_-shared/lib/python3.6/site-packages/pdm/pep517/base.py", line 277, in format_setup_py
pdm.termui:     "from {} import build\n".format(meta.build.split(".")[0]),
pdm.termui: AttributeError: 'dict' object has no attribute 'split'
  ✖ Install baize 0.17.1 failed
```

https://github.com/abersheeran/baize/runs/7168547020?check_suite_focus=true

pdm 2.0.0b1

**Comment by frostming** [comment]
Python 3.6 **can't** pull the latest `pdm-pep517`, and can't parse the new build configuration.

So you need to either use the old build configuration, or drop the Python 3.6 support.

**Comment by frostming** [comment]
Hmm, there is a default value issue that prevents this, will fix it soon

**Comment by frostming** [comment]
Fixed in 1.0.1, the above two options still apply
